library/blocks/block-extensions/src/main/java/org/quiltmc/qsl/block/extensions/api/QuiltBlockSettings.java

```diff
@@ -0,0 +1,308 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.block.extensions.api;
+
+import net.minecraft.block.*;
+import net.minecraft.entity.EntityType;
+import net.minecraft.sound.BlockSoundGroup;
+import net.minecraft.util.DyeColor;
+import net.minecraft.util.Identifier;
+import org.quiltmc.qsl.block.extensions.api.data.BlockDataKey;
+import org.quiltmc.qsl.block.extensions.impl.BlockSettingsInternals;
+import org.quiltmc.qsl.block.extensions.mixin.AbstractBlockAccessor;
+import org.quiltmc.qsl.block.extensions.mixin.AbstractBlockSettingsAccessor;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.function.ToIntFunction;
+
+/**
+ * An extended variant of the {@link AbstractBlock.Settings} class, which provides extra methods for customization.
+ */
+public class QuiltBlockSettings extends AbstractBlock.Settings {
+	protected QuiltBlockSettings(Material material, Function<BlockState, MapColor> mapColorProvider) {
+		super(material, mapColorProvider);
+	}
+
+	protected QuiltBlockSettings(Material material, MapColor mapColor) {
+		super(material, mapColor);
+	}
+
+	protected QuiltBlockSettings(AbstractBlock.Settings settings) {
+		super(((AbstractBlockSettingsAccessor) settings).getMaterial(), ((AbstractBlockSettingsAccessor) settings).getMapColorProvider());
+
+		// [VanillaCopy] AbstractBlock.Settings#copy(AbstractBlock.Settings)
+		var thisAccessor = (AbstractBlockSettingsAccessor) this;
+		var otherAccessor = (AbstractBlockSettingsAccessor) settings;
+
+		thisAccessor.setMaterial(otherAccessor.getMaterial());
+		this.hardness(otherAccessor.getHardness());
+		this.resistance(otherAccessor.getResistance());
+		this.collidable(otherAccessor.getCollidable());
+		thisAccessor.setRandomTicks(otherAccessor.getRandomTicks());
+		this.luminance(otherAccessor.getLuminance());
+		thisAccessor.setMapColorProvider(otherAccessor.getMapColorProvider());
+		this.sounds(otherAccessor.getSoundGroup());
+		this.slipperiness(otherAccessor.getSlipperiness());
+		this.velocityMultiplier(otherAccessor.getVelocityMultiplier());
+		thisAccessor.setDynamicBounds(otherAccessor.getDynamicBounds());
+		thisAccessor.setOpaque(otherAccessor.getOpaque());
+		thisAccessor.setIsAir(otherAccessor.getIsAir());
+		thisAccessor.setToolRequired(otherAccessor.isToolRequired());
+
+		// also copy extra data
+		var thisInternals = (BlockSettingsInternals) this;
+		var otherInternals = (BlockSettingsInternals) this;
+
+		Map<BlockDataKey<?>, Object> settingsMap = otherInternals.qsl$getSettingsMap();
+		if (settingsMap != null)
+			thisInternals.qsl$setSettingsMap(new HashMap<>(settingsMap));
+	}
```

Comment:
Does this copy the other settings' loot table ID?

Reply:
Nope. Neither Vanilla nor Fabric bother with it, I'll add it now.

Reply:
This also doesn't copy the `jumpVelocityMultiplier` and all the various `*Predicate`s